WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -328,6 +342,44 @@ private static bool IsIReferenceArray(Type implementationType)
             return CreateFactoryForImplementationType(runtimeClassName, implementationType);
         }
 
+        internal static string GetRuntimeClassForTypeCreation(IInspectable inspectable, Type staticallyDeterminedType)
+        {
+            string runtimeClassName = inspectable.GetRuntimeClassName(noThrow: true);

```

Comment:
If it throws, we return null and caller handles it by just using the iinspecatable rather than casting it to a type.  This is what we did for the .NET 5 projection before.  Looks like .NET standard 2 projection might have thrown before in this case, but I don't think that was intentional as they are also have logic to return iinspectable but runtimeclass name is null.